### ISSUE TITLE
Auto-login after email validation and redirection to 'fos_user_profile_edit' path

### ISSUE THREAD

**Issue by bhss** [FriendsOfSymfony/FOSUserBundle#2586]
We're building a business app from the ground up in Symfony 3, and I've run into a bit of a snag with the user validation flow: after the user creates an account they will receive a validation email, then(after clicking the confirmation/validation link) they should be automatically authentificated and redirected to the edit profile.
But unfortunately they are redirected to the login page without any authentification.

Anyone had any experience with this, or able to point me in the right direction?

**Comment by Laurent3170** [comment]
You should ask on StackOverflow as it's not an issue, but I think you should build an EventListener to catch `FOSUserEvents::REGISTRATION_CONFIRMED`
```
use FOS\UserBundle\FOSUserEvents;
use FOS\UserBundle\Event\FilterUserResponseEvent;
use FOS\UserBundle\Event\FormEvent;
use Symfony\Component\EventDispatcher\EventSubscriberInterface;

class RegistrationListener implements EventSubscriberInterface
{
    public static function getSubscribedEvents()
    {
        return array(
            FOSUserEvents::REGISTRATION_CONFIRMED => 'onRegistrationConfirmed',
        );
    }
  
    public function onRegistrationConfirmed(FilterUserResponseEvent $event)
    {
        //Redirects to edit profile
        $this->redirectToRoute('fos_user_profile_edit');
    }
}

**Comment by bhss** [comment]
Does by this way, the user gonna authentificate  automaticlly ?

**Comment by Laurent3170** [comment]
Yes the function `confirmAction()` loads the user, but the `RedirectResponse()` in the code above is not taken into account, so it doesn't work...
We should update the Event but there is no function `setResponse()` to do so, neither `getUser()` and `getRequest()` to re-build the event.
So the solution, as indicated in https://stackoverflow.com/questions/16427267/fosuserbundle-redirect-the-user-after-register-with-eventlistener is to catch `FOSUserEvents::REGISTRATION_CONFIRM` and then update the Response in the event.
